controllers/logicalvolume_controller.go

```diff
@@ -184,29 +189,60 @@ func (r *LogicalVolumeReconciler) createLV(ctx context.Context, log logr.Logger,
 
 	resp, err := lvService.CreateLV(ctx, &proto.CreateLVRequest{Name: string(lv.UID), SizeGb: uint64(reqBytes >> 30)})
 	if err != nil {
-		s, ok := status.FromError(err)
-		if !ok {
-			log.Error(err, err.Error())
-			err := r.updateStatusWithError(ctx, log, lv, codes.Internal, err.Error())
-			return err
-		}
-		log.Error(err, s.Message())
-		err := r.updateStatusWithError(ctx, log, lv, s.Code(), s.Message())
+		code, message := extractFromError(err)
+		log.Error(err, message)
+		err := r.updateStatusWithError(ctx, log, lv, code, message)
 		return err
 	}
 
 	lv2 := lv.DeepCopy()
 	lv2.Status.VolumeID = resp.Volume.Name
+	lv2.Status.CurrentSize = resource.NewQuantity(reqBytes, resource.BinarySI)
 	patch := client.MergeFrom(lv)
 	if err := r.Status().Patch(ctx, lv2, patch); err != nil {
-		log.Error(err, "failed to update VolumeID", "name", lv.Name, "uid", lv.UID)
+		log.Error(err, "failed to update status", "name", lv.Name, "uid", lv.UID)
 		return err
 	}
 
 	log.Info("created new LV", "name", lv2.Name, "uid", lv2.UID, "status.volumeID", lv2.Status.VolumeID)
 	return nil
 }
 
+func (r *LogicalVolumeReconciler) expandLV(ctx context.Context, log logr.Logger, lv *topolvmv1.LogicalVolume,
+	vgService proto.VGServiceClient, lvService proto.LVServiceClient) error {
+	if lv.Status.CurrentSize == nil {
+		log.Info(".status.currentSize is expected but not set", "name", lv.Name)
+		// return nil not to be requeued
+		return nil
+	}
```

Comment:
I do not understand these logics.
When upgrading from v0.3, `Status.CurrentSize` should always nil, right?
So, what does this check do?

Reply:
When upgrading from v0.3, topolvm-controller first fills `Status.CurrentSize` with the value of `Spec.Size` in `ControllerExpandVolume()`, and then updates `Spec.Size` to call this `expandLV()`.
So `Status.CurrentSize` should not be nil here.